From d3d94ef4a54f672b774bca4f8d926ff76ce9582a Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Fri, 26 Oct 2012 15:49:22 +0300
Subject: [PATCH 3/4] OF: Enable OF_DYNAMIC

---
 drivers/of/Kconfig |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/of/Kconfig b/drivers/of/Kconfig
index dfba3e6..00e218c 100644
--- a/drivers/of/Kconfig
+++ b/drivers/of/Kconfig
@@ -38,7 +38,9 @@ config OF_PROMTREE
 # Hardly any platforms need this.  It is safe to select, but only do so if you
 # need it.
 config OF_DYNAMIC
-	bool
+	bool "Enable dynamic update of Device Tree"
+	help
+	  Enable this to display changes in the device tree at runtime.
 
 config OF_ADDRESS
 	def_bool y
-- 
1.7.7.6

