From fcf285a211940a50b4e5a7c9f02de33279d847af Mon Sep 17 00:00:00 2001
From: Joel A Fernandes <joelagnel@ti.com>
Date: Tue, 20 Nov 2012 14:14:27 -0600
Subject: [PATCH 5/6] video/Kconfig+Makefile: Add new Kconfig for old driver

Signed-off-by: Joel A Fernandes <joelagnel@ti.com>
---
 drivers/video/Kconfig  |    2 ++
 drivers/video/Makefile |    2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/video/Kconfig b/drivers/video/Kconfig
index e7868d8..3dc4ae8 100644
--- a/drivers/video/Kconfig
+++ b/drivers/video/Kconfig
@@ -2433,6 +2433,8 @@ source "drivers/video/omap2/Kconfig"
 source "drivers/video/exynos/Kconfig"
 source "drivers/video/backlight/Kconfig"
 
+source "drivers/video/nxp/Kconfig"
+
 if VT
 	source "drivers/video/console/Kconfig"
 endif
diff --git a/drivers/video/Makefile b/drivers/video/Makefile
index 0e13296..10e0642 100644
--- a/drivers/video/Makefile
+++ b/drivers/video/Makefile
@@ -168,3 +168,5 @@ obj-$(CONFIG_FB_VIRTUAL)          += vfb.o
 
 #video output switch sysfs driver
 obj-$(CONFIG_VIDEO_OUTPUT_CONTROL) += output.o
+
+obj-$(CONFIG_NXP_TDA998X_OLD) 	  += nxp/
-- 
1.7.9.5

