From 5c2c648a72ebb0306fdcd1075380751b7aa3e09b Mon Sep 17 00:00:00 2001
From: Joel A Fernandes <joelagnel@ti.com>
Date: Mon, 3 Dec 2012 11:08:39 -0600
Subject: [PATCH 7/8] tda driver: enable 1280x720

Signed-off-by: Joel A Fernandes <joelagnel@ti.com>
---
 drivers/video/nxp/tda998x.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/nxp/tda998x.c b/drivers/video/nxp/tda998x.c
index 2864221..de9ce6d 100755
--- a/drivers/video/nxp/tda998x.c
+++ b/drivers/video/nxp/tda998x.c
@@ -1034,9 +1034,9 @@ static int hdmi_tx_init(tda_instance *this)
    /*    this->tda.setio.video_out.format = TMDL_HDMITX_VFMT_31_1920x1080p_50Hz; */
    /*    this->tda.setio.video_out.format = TMDL_HDMITX_VFMT_PC_640x480p_60Hz; */
    /*    this->tda.setio.video_out.format = TMDL_HDMITX_VFMT_PC_640x480p_72Hz; */
-   // this->tda.setio.video_out.format = TMDL_HDMITX_VFMT_04_1280x720p_60Hz; 
+   this->tda.setio.video_out.format = TMDL_HDMITX_VFMT_04_1280x720p_60Hz; 
    /*    this->tda.setio.video_out.format = TMDL_HDMITX_VFMT_19_1280x720p_50Hz; */
-   this->tda.setio.video_out.format = TMDL_HDMITX_VFMT_02_720x480p_60Hz;
+   // this->tda.setio.video_out.format = TMDL_HDMITX_VFMT_02_720x480p_60Hz;
 
    this->tda.setio.video_in.mode = TMDL_HDMITX_VINMODE_RGB444;
    /*    this->tda.setio.video_in.mode = TMDL_HDMITX_VINMODE_CCIR656; */
-- 
1.7.9.5

