From ef496dda5c4c3f07f571ddcfaa3814a7c7047177 Mon Sep 17 00:00:00 2001
From: Joel A Fernandes <joelagnel@ti.com>
Date: Mon, 3 Dec 2012 11:09:29 -0600
Subject: [PATCH 8/8] Makefile: Disable CEC

Signed-off-by: Joel A Fernandes <joelagnel@ti.com>
---
 drivers/video/nxp/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/nxp/Makefile b/drivers/video/nxp/Makefile
index f583226..05dc816 100755
--- a/drivers/video/nxp/Makefile
+++ b/drivers/video/nxp/Makefile
@@ -14,7 +14,7 @@ TDA_PLATFORM := ZOOMII
 
 #TDA_HDCP := 0
 TDA_HDCP := TMFL_HDCP_SUPPORT
-TDA_CEC := TDA9950
+#TDA_CEC := TDA9950
 
 # add this if INTERRUPT is wired, otherwise polling with timer is used
 #EXTRA_CFLAGS += -DIRQ
-- 
1.7.9.5

