From f9e8b39d6e660288a820703eadc8107b1db47022 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 29 Feb 2012 16:32:49 -0800
Subject: [PATCH 72/72] Linux 3.2.9

---
 Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7df8a84..5f1739b 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 2
-SUBLEVEL = 8
+SUBLEVEL = 9
 EXTRAVERSION =
 NAME = Saber-toothed Squirrel
 
-- 
1.7.9.4

