From 0db9ffe56bd7dc92b88293b07c18ae9b48663062 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 24 Oct 2012 11:34:54 +0200
Subject: [PATCH 20/20] f2fs: compile fix

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 fs/f2fs/segment.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/fs/f2fs/segment.c b/fs/f2fs/segment.c
index 57d0931..5bab838 100644
--- a/fs/f2fs/segment.c
+++ b/fs/f2fs/segment.c
@@ -12,6 +12,7 @@
 #include <linux/f2fs_fs.h>
 #include <linux/bio.h>
 #include <linux/blkdev.h>
+#include <linux/vmalloc.h>
 
 #include "f2fs.h"
 #include "segment.h"
-- 
1.7.7.6

