From 5e2dfb44f5f13fd1b0c14f8841619b39d21d9587 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Wed, 17 Oct 2012 15:43:55 +0300
Subject: [PATCH 4/7] da8xx: Add CDTech_S035Q01 panel (used by LCD3 bone cape)

---
 drivers/video/da8xx-fb.c |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/drivers/video/da8xx-fb.c b/drivers/video/da8xx-fb.c
index 5d16494..4713b83 100644
--- a/drivers/video/da8xx-fb.c
+++ b/drivers/video/da8xx-fb.c
@@ -274,6 +274,21 @@ static struct fb_videomode known_lcd_panels[] = {
 		.sync           = FB_SYNC_HOR_HIGH_ACT | FB_SYNC_VERT_HIGH_ACT,
 		.flag           = 0,
 	},
+	[4] = {
+		/* CDTech S035Q01 */
+		.name 		= "CDTech_S035Q01",
+		.xres 		= 320,
+		.yres		= 240,
+		.pixclock	= 8000000,
+		.left_margin    = 21,
+		.right_margin   = 58,
+		.upper_margin   = 11,
+		.lower_margin   = 23,
+		.hsync_len      = 47,
+		.vsync_len      = 2,
+		.sync           = FB_SYNC_HOR_HIGH_ACT | FB_SYNC_VERT_HIGH_ACT,
+		.flag           = 0,
+	},
 };
 
 /* Enable the Raster Engine of the LCD Controller */
-- 
1.7.7.6

