From a3992da7e6428409ef67e9c6a490dd7aa8a64152 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 6 Mar 2014 16:05:04 -0600
Subject: [PATCH 4/4] build: capes-one-layer

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index d57c1a6..bfb75e4 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -198,6 +198,11 @@ dtb-$(CONFIG_ARCH_OMAP2PLUS) += omap2420-h4.dtb \
 	am335x-evmsk.dtb \
 	am335x-bone.dtb \
 	am335x-boneblack.dtb \
+	am335x-boneblack-default.dtb \
+	am335x-boneblack-ttyO1.dtb \
+	am335x-boneblack-ttyO2.dtb \
+	am335x-boneblack-ttyO4.dtb \
+	am335x-boneblack-cape-bone-argus.dtb \
 	am335x-nano.dtb \
 	am335x-base0033.dtb \
 	am3517-evm.dtb \
-- 
1.9.0

