diff --git a/arch/arm/boot/dts/am335x-bone.dts b/arch/arm/boot/dts/am335x-bone.dts
index dee02b3..94188bb 100644
--- a/arch/arm/boot/dts/am335x-bone.dts
+++ b/arch/arm/boot/dts/am335x-bone.dts
@@ -39,7 +39,7 @@
 				0x194 0x33	/* mcasp0_fsx.spi1_d0, INPUT_PULLUP | MODE3 */
 				0x198 0x13	/* mcasp0_axr0.spi1_d1, OUTPUT_PULLUP | MODE3 */
 				0x19c 0x13	/* mcasp0_ahclkr.spi1_cs0, OUTPUT_PULLUP | MODE3 */
-				0x48 0x6        /* gpmc_a2.ehrpwm1A, MODE6 */
+				0x4c 0x6        /* gpmc_a3.ehrpwm1B, MODE6 */
 			>;
 		};
 		lcd_pins: pinmux_lcd_pins {
@@ -56,7 +56,31 @@
 		};
 		pruss_led_pins: pinmux_pruss_led_pins {
 			pinctrl-single,pins = <
-				0x0a0 0x15	/* lcd_data0.pr1_pru1_pru_r30_0, OUTPUT_PULLUP | MODE5 */
+				0x018 0x47 /* P8_3 gpmc_ad6.gpio1_6 OMAP_PIN_OUTPUT | OMAP_MUX_MODE7 */
+				0x008 0x57 /* P8_5 gpmc_ad2.gpio1_2 OMAP_PIN_OUTPUT | OMAP_MUX_MODE7 */
+				0x0c 0x37  /* gpmc_ad3.gpio1_3, OMAP_PIN_INPUT_PULLUP | OMAP_MUX_MODE7 - w1-gpio */
+				0x084 0x05 /* P8_20 gpmc_csn2.pr1_pru1_pru_r30_13 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x080 0x05 /* P8_21 gpmc_csn1.pr1_pru1_pru_r30_12 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x000 0x07 /* P8_25 gpmc_ad0.gpio1_0 OMAP_PIN_OUTPUT | OMAP_MUX_MODE7 */
+				0x0e0 0x05 /* P8_27 lcd_vsync.pr1_pru1_pru_r30_8 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x0e8 0x05 /* P8_28 lcd_pclk.pr1_pru1_pru_r30_10 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x0e4 0x05 /* P8_29 lcd_hsync.pr1_pru1_pru_r30_9 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x0ec 0x05 /* P8_30 lcd_ac_bias_en.pr1_pru1_pru_r30_11 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x0d8 0x27 /* P8_31 lcd_data14.gpio0_10 OMAP_PIN_INPUT_PULLDOWN | OMAP_MUX_MODE7 */
+				0x0dc 0x27 /* P8_32 lcd_data15.gpio0_11 OMAP_PIN_INPUT_PULLDOWN | OMAP_MUX_MODE7 */
+				0x0d4 0x27 /* P8_33 lcd_data13.gpio0_9 OMAP_PIN_INPUT_PULLDOWN | OMAP_MUX_MODE7 */
+				0x0d0 0x27 /* P8_35 lcd_data12.gpio0_8 OMAP_PIN_INPUT_PULLDOWN | OMAP_MUX_MODE7 */
+				0x0c8 0x02 /* P8_36 lcd_data10.ehrpwm1A OMAP_PIN_OUTPUT | OMAP_MUX_MODE2 */
+				0x0c0 0x27 /* P8_37 lcd_data8.gpio2_14 OMAP_PIN_INPUT_PULLDOWN | OMAP_MUX_MODE7 */
+				0x0c4 0x27 /* P8_38 lcd_data9.gpio2_15 OMAP_PIN_INPUT_PULLDOWN | OMAP_MUX_MODE7 */
+				0x0b8 0x05 /* P8_39 lcd_data6.pr1_pru1_pru_r30_6 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x0bc 0x05 /* P8_40 lcd_data7.pr1_pru_pru1_r30_7 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x0b0 0x05 /* P8_41 lcd_data4.pr1_pru1_pru_r30_4 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x0b4 0x05 /* P8_42 lcd_data5.pr1_pru1_pru_r30_5 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x0a8 0x05 /* P8_43 lcd_data2.pr1_pru1_pru_r30_2 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x0ac 0x05 /* P8_44 lcd_data3.pr1_pru1_pru_r30_3 OMAP_PIN_OUTPUT | OMAP_MUX_MODE5 */
+				0x0a0 0x03 /* P8_45 lcd_data0.ehrpwm2A OMAP_PIN_OUTPUT | OMAP_MUX_MODE3 */
+				0x0a4 0x03 /* P8_46 lcd_data1.ehrpwm2B OMAP_PIN_OUTPUT | OMAP_MUX_MODE3 */
 			>;
 		};
 	};
@@ -101,11 +125,35 @@
 
 	st7735fb-bl {
 			compatible	= "pwm-backlight"; 
-			pwms = <&ehrpwm1 0 500000 0>;
+			pwms = <&ehrpwm1 1 500000 0>;
 			pwm-names = "st7735fb";
 			brightness-levels = <0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100>;
 			default-brightness-level = <50>; /* index to the array above */ 
 		};
+
+	onewire@0 {
+		compatible = "w1-gpio";
+		gpios = <&gpio1 3 0>;
+	};
+
+	pwm_test0: ehrpwm1A {
+		compatible	= "pwm_test";
+		pwms = <&ehrpwm1 0 5000000 0>;
+		pwm-names = "ehrpwm.1";
+	};
+
+	pwm_test1: ehrpwm2A {
+		compatible	= "pwm_test";
+		pwms = <&ehrpwm2 0 5000000 0>;
+		pwm-names = "ehrpwm.2";
+	};
+
+	pwm_test2: ehrpwm2B {
+		compatible	= "pwm_test";
+		pwms = <&ehrpwm2 1 5000000 0>;
+		pwm-names = "ehrpwm.2";
+	};
+
 };
 
 &i2c0 {
@@ -242,3 +290,6 @@
 	pinctrl-0 = <&pruss_led_pins>;
 };
 
+&ehrpwm2 {
+	status = "okay";
+};
