From ad414d57ad5c1103b6907ccc6632b51bfba4b27f Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 10 Jul 2013 14:32:25 +0200
Subject: [PATCH 20/23] firmware: remove rule for cape-bone-adafruit-lcd-00A0

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 firmware/Makefile |    1 -
 1 file changed, 1 deletion(-)

diff --git a/firmware/Makefile b/firmware/Makefile
index e6d178e..cdfc800 100644
--- a/firmware/Makefile
+++ b/firmware/Makefile
@@ -140,7 +140,6 @@ fw-shipped-$(CONFIG_YAM) += yam/1200.bin yam/9600.bin
 fw-shipped-$(CONFIG_CAPE_BEAGLEBONE) += \
 	cape-bone-2g-emmc1.dtbo \
 	cape-bone-adafruit-lcd-00A0.dtbo \
-	cape-bone-adafruit-rtc-00A0.dtbo \
 	cape-bone-dvi-00A0.dtbo \
 	cape-bone-dvi-00A1.dtbo \
 	cape-bone-dvi-00A2.dtbo \
-- 
1.7.10.4

