From b81c7df5d7781ea99781aab9c72ce8cc02dba985 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 12 Dec 2012 16:15:04 -0600
Subject: [PATCH] merge cleanup ispvideo.c remove includes

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/media/platform/omap3isp/ispvideo.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/media/platform/omap3isp/ispvideo.c b/drivers/media/platform/omap3isp/ispvideo.c
index 3311d6b..293e380 100644
--- a/drivers/media/platform/omap3isp/ispvideo.c
+++ b/drivers/media/platform/omap3isp/ispvideo.c
@@ -35,9 +35,6 @@
 #include <linux/vmalloc.h>
 #include <media/v4l2-dev.h>
 #include <media/v4l2-ioctl.h>
-#include <plat/iommu.h>
-#include <plat/iovmm.h>
-#include <plat/omap-pm.h>
 
 #include "ispvideo.h"
 #include "isp.h"
-- 
1.7.10.4

