From dc9670576bd8459e8c61162f5475fbf0304828b9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Mon, 21 May 2012 17:01:23 +0200
Subject: [PATCH 16/16] remove debug msgs due to missing in_interrupt
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>
---
 drivers/net/wireless/libertas/defs.h | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/net/wireless/libertas/defs.h b/drivers/net/wireless/libertas/defs.h
index ab966f0..a80b40e 100644
--- a/drivers/net/wireless/libertas/defs.h
+++ b/drivers/net/wireless/libertas/defs.h
@@ -46,14 +46,7 @@
 
 extern unsigned int lbs_debug;
 
-#ifdef DEBUG
-#define LBS_DEB_LL(grp, grpnam, fmt, args...) \
-do { if ((lbs_debug & (grp)) == (grp)) \
-  printk(KERN_DEBUG DRV_NAME grpnam "%s: " fmt, \
-         in_interrupt() ? " (INT)" : "", ## args); } while (0)
-#else
 #define LBS_DEB_LL(grp, grpnam, fmt, args...) do {} while (0)
-#endif
 
 #define lbs_deb_enter(grp) \
   LBS_DEB_LL(grp | LBS_DEB_ENTER, " enter", "%s()\n", __func__);
-- 
1.8.1.4

