From 00af5cb98c9a7a213dbb1e90f8ef318e09a675b5 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Fri, 28 Dec 2012 20:23:17 +0200
Subject: [PATCH 022/184] firmware: update .gitignore with dtbo objects

Ignore generated dtbo files.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 firmware/.gitignore |    1 +
 1 file changed, 1 insertion(+)

diff --git a/firmware/.gitignore b/firmware/.gitignore
index d9c6901..21e3640 100644
--- a/firmware/.gitignore
+++ b/firmware/.gitignore
@@ -4,3 +4,4 @@
 *.csp
 *.dsp
 ihex2fw
+*.dtbo
-- 
1.7.10.4

