From f33c00b94d47bac5f521ddf1b106004709bcb170 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Thu, 17 Jan 2013 10:26:39 +0100
Subject: [PATCH 076/184] firmware: fix dvi a1 target

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 firmware/Makefile |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/firmware/Makefile b/firmware/Makefile
index 2f14ea4..0bcacba 100644
--- a/firmware/Makefile
+++ b/firmware/Makefile
@@ -139,7 +139,8 @@ fw-shipped-$(CONFIG_YAM) += yam/1200.bin yam/9600.bin
 # all the generic capes
 fw-shipped-$(CONFIG_CAPE_BEAGLEBONE) += \
 	cape-bone-2g-emmc1.dtbo \
-	cape-bone-dvi-00A0.dtbo capes/cape-bone-dvi-00A1.dtbo \
+	cape-bone-dvi-00A0.dtbo \
+	cape-bone-dvi-00A1.dtbo \
 	cape-bone-dvi-00A2.dtbo \
 	cape-bone-lcd3-00A0.dtbo \
 	cape-bone-lcd3-00A2.dtbo
-- 
1.7.10.4

