From 7e3b754f4a8c7ae50a7fa703756ac93840aa5030 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Fri, 5 Apr 2013 12:07:30 +0200
Subject: [PATCH 157/184] cape-bone-hexy: add iio helper

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 firmware/capes/cape-bone-hexy-00A0.dts |   30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/firmware/capes/cape-bone-hexy-00A0.dts b/firmware/capes/cape-bone-hexy-00A0.dts
index abc1255..0e90904 100644
--- a/firmware/capes/cape-bone-hexy-00A0.dts
+++ b/firmware/capes/cape-bone-hexy-00A0.dts
@@ -157,4 +157,34 @@
 			};
 		};
 	};
+
+	fragment@6 {
+		target = <&ocp>;
+		__overlay__ {
+			/* avoid stupid warning */
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			tscadc {
+				compatible = "ti,ti-tscadc";
+				reg = <0x44e0d000 0x1000>;
+
+				interrupt-parent = <&intc>;
+				interrupts = <16>;
+				ti,hwmods = "adc_tsc";
+				status = "okay";
+
+				adc {
+					ti,adc-channels = <0 1 2 3 4 5 6 7>;
+				};
+			};
+
+			test_helper: helper {
+				compatible = "bone-iio-helper";
+				vsense-name  = "AIN0", "AIN1", "AIN2", "AIN3", "AIN4", "AIN5", "AIN6", "AIN7";
+				vsense-scale = <100     100     100     100     100     100     100     100>;
+				status = "okay";
+			};
+		};
+	};
 };
-- 
1.7.10.4

