From 638389b32b46e5da3f1788c6d1b2ea27dc4ab481 Mon Sep 17 00:00:00 2001
From: Charles Steinkuehler <charles@steinkuehler.net>
Date: Sun, 26 Jan 2014 16:17:09 +0100
Subject: [PATCH] Add dir-changeable property to gpio-of-helper

Signed-off-by: Charles Steinkuehler <charles@steinkuehler.net>
---
 drivers/gpio/gpio-of-helper.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpio/gpio-of-helper.c b/drivers/gpio/gpio-of-helper.c
index 16a607f..1c289b2 100644
--- a/drivers/gpio/gpio-of-helper.c
+++ b/drivers/gpio/gpio-of-helper.c
@@ -181,6 +181,8 @@ gpio_of_entry_create(struct gpio_of_helper_info *info,
 				req_flags |= GPIOF_OUT_INIT_LOW;
 			break;
 	}
+	if (of_property_read_bool(node, "dir-changeable"))
+		req_flags |= GPIOF_EXPORT_CHANGEABLE;
 
 	/* request the gpio */
 	err = devm_gpio_request_one(dev, gpio, req_flags, name);
-- 
1.8.5.3

