From 3dc6e3f4eb7df083896a6bc9bca0a1b745f3dc3d Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 2 Jan 2014 10:11:58 -0600
Subject: [PATCH 21/21] Fix aspect ratio issue of 720p in MT9M114 camera cape

Signed-off-by: vvdn <anithamol.benny@vvdntech.com>
---
 drivers/media/i2c/soc_camera/mt9m114.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/drivers/media/i2c/soc_camera/mt9m114.c b/drivers/media/i2c/soc_camera/mt9m114.c
index 6eab204..9df2119 100644
--- a/drivers/media/i2c/soc_camera/mt9m114.c
+++ b/drivers/media/i2c/soc_camera/mt9m114.c
@@ -411,22 +411,22 @@ static const struct mt9m114_reg mt9m114_regs_wvga[] = {
 
 static const struct mt9m114_reg mt9m114_regs_720p[] = {
 	{ MT9M114_LOGICAL_ADDRESS_ACCESS,                0x1000, 2 },
-	{ MT9M114_CAM_SENSOR_CFG_Y_ADDR_START,           0x0004, 2 },
+	{ MT9M114_CAM_SENSOR_CFG_Y_ADDR_START,           0x007C, 2 },
 	{ MT9M114_CAM_SENSOR_CFG_X_ADDR_START,           0x0004, 2 },
-	{ MT9M114_CAM_SENSOR_CFG_Y_ADDR_END,             0x03CB, 2 },
+	{ MT9M114_CAM_SENSOR_CFG_Y_ADDR_END,             0x0353, 2 },
 	{ MT9M114_CAM_SENSOR_CFG_X_ADDR_END,             0x050B, 2 },
 	{ MT9M114_CAM_SENSOR_CFG_ROW_SPEED,              0x0001, 2 },
-	{ MT9M114_CAM_SENSOR_CFG_FINE_INTEG_TIME_MIN,    0x00DB, 2 },
-	{ MT9M114_CAM_SENSOR_CFG_FINE_INTEG_TIME_MAX,    0x05B3, 2 },
-	{ MT9M114_CAM_SENSOR_CFG_FRAME_LENGTH_LINES,     0x03EE, 2 },
-	{ MT9M114_CAM_SENSOR_CFG_LINE_LENGTH_PCK,        0x0636, 2 },
+	{ MT9M114_CAM_SENSOR_CFG_FINE_INTEG_TIME_MIN,    0x00A0, 2 },
+	{ MT9M114_CAM_SENSOR_CFG_FINE_INTEG_TIME_MAX,    0x03F7, 2 },
+	{ MT9M114_CAM_SENSOR_CFG_FRAME_LENGTH_LINES,     0x02D5, 2 },
+	{ MT9M114_CAM_SENSOR_CFG_LINE_LENGTH_PCK,        0x060E, 2 },
 	{ MT9M114_CAM_SENSOR_CFG_FINE_CORRECTION,        0x0060, 2 },
-	{ MT9M114_CAM_SENSOR_CFG_CPIPE_LAST_ROW,         0x03C3, 2 },
+	{ MT9M114_CAM_SENSOR_CFG_CPIPE_LAST_ROW,         0x02D3, 2 },
 	{ MT9M114_CAM_SENSOR_CFG_REG_0_DATA,             0x0020, 2 },
 	{ MT9M114_CAM_CROP_WINDOW_XOFFSET,               0x0000, 2 },
 	{ MT9M114_CAM_CROP_WINDOW_YOFFSET,               0x0000, 2 },
 	{ MT9M114_CAM_CROP_WINDOW_WIDTH,                 0x0500, 2 },
-	{ MT9M114_CAM_CROP_WINDOW_HEIGHT,                0x03C0, 2 },
+	{ MT9M114_CAM_CROP_WINDOW_HEIGHT,                0x02D0, 2 },
 	{ MT9M114_CAM_CROP_CROPMODE,                     0x03,   1 },
 	{ MT9M114_CAM_OUTPUT_WIDTH,                      0x0500, 2 },
 	{ MT9M114_CAM_OUTPUT_HEIGHT,                     0x02D0, 2 },
@@ -437,8 +437,8 @@ static const struct mt9m114_reg mt9m114_regs_720p[] = {
 	{ MT9M114_CAM_STAT_AWB_CLIP_WINDOW_YEND,         0x02CF, 2 },
 	{ MT9M114_CAM_STAT_AE_INITIAL_WINDOW_XSTART,     0x0000, 2 },
 	{ MT9M114_CAM_STAT_AE_INITIAL_WINDOW_YSTART,     0x0000, 2 },
-	{ MT9M114_CAM_STAT_AE_INITIAL_WINDOW_XEND,       0x00FF, 2 },
-	{ MT9M114_CAM_STAT_AE_INITIAL_WINDOW_YEND,       0x008F, 2 },
+	{ MT9M114_CAM_STAT_AE_INITIAL_WINDOW_XEND,       0x009F, 2 },
+	{ MT9M114_CAM_STAT_AE_INITIAL_WINDOW_YEND,       0x005F, 2 },
 };
 
 static const struct mt9m114_format {
-- 
1.8.5.2

