From 4287973d87c99fe966126998db3ee42ed7593513 Mon Sep 17 00:00:00 2001
From: Matt Ranostay <mranostay@gmail.com>
Date: Sat, 21 Dec 2013 22:27:48 -0800
Subject: [PATCH 25/40] capes: reorder from GRB to RGB for Lighting cape

Signed-off-by: Matt Ranostay <mranostay@gmail.com>
---
 drivers/remoteproc/pru_rproc.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/remoteproc/pru_rproc.c b/drivers/remoteproc/pru_rproc.c
index 36d2ad5..bc3900f 100644
--- a/drivers/remoteproc/pru_rproc.c
+++ b/drivers/remoteproc/pru_rproc.c
@@ -2444,9 +2444,9 @@ static long ioctl_spiwrite_burst(struct pruproc *pp, struct spi_ioc_transfer *sp
 	memset(array->data, 0, MAX_SLOTS * 4);
 
 	for (i = 0; i < (spi->len / 3); i++) {
-		u32 val = *(d++) << 0;
-		val |= *(d++) << 8;
-		val |= *(d++) << 16;
+		u32 val = *(d++) << 8; /* Red */
+		val |= *(d++) << 16;    /* Green */
+		val |= *(d++) << 0;   /* Blue */
 
 		array->data[slot] = val;
 		if (++slot > max_slots) {
-- 
1.9.0

