From 840bd524d62592a5284809ee5535726e8ac6f4d0 Mon Sep 17 00:00:00 2001
From: Matt Porter <mporter@ti.com>
Date: Fri, 19 Oct 2012 11:24:40 -0400
Subject: [PATCH 31/31] proposed probe fix - works for me on evm

Signed-off-by: Matt Porter <mporter@ti.com>
---
 arch/arm/common/edma.c |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/common/edma.c b/arch/arm/common/edma.c
index b761b7a..6ed394f 100644
--- a/arch/arm/common/edma.c
+++ b/arch/arm/common/edma.c
@@ -1598,6 +1598,8 @@ static struct of_dma_filter_info edma_filter_info = {
 static int __init edma_probe(struct platform_device *pdev)
 {
 	struct edma_soc_info	**info = pdev->dev.platform_data;
+	struct edma_soc_info	*ninfo[EDMA_MAX_CC] = {NULL, NULL};
+	struct edma_soc_info	tmpinfo;
 	s8			(*queue_priority_mapping)[2];
 	s8			(*queue_tc_mapping)[2];
 	int			i, j, off, ln, found = 0;
@@ -1614,15 +1616,13 @@ static int __init edma_probe(struct platform_device *pdev)
 	char			irq_name[10];
 	struct device_node	*node = pdev->dev.of_node;
 	struct device		*dev = &pdev->dev;
-	struct edma_soc_info	*pdata;
 	int			ret;
 
 	if (node) {
-		pdata = devm_kzalloc(dev,
-				     sizeof(struct edma_soc_info),
-				     GFP_KERNEL);
-		edma_of_parse_dt(dev, node, pdata);
-		info = &pdata;
+		info = ninfo;
+		edma_of_parse_dt(dev, node, &tmpinfo);
+		info[0] = &tmpinfo;
+
 		dma_cap_set(DMA_SLAVE, edma_filter_info.dma_cap);
 		of_dma_controller_register(dev->of_node,
 					   of_dma_simple_xlate,
-- 
1.7.7.6

