From 0a4a06fdba20b76d03dde619a4dec0e88cdad6c3 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Fri, 7 Oct 2011 15:29:28 +0200
Subject: [PATCH 01/26] f_rndis: HACK around undefined variables

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 drivers/usb/gadget/f_rndis.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/f_rndis.c b/drivers/usb/gadget/f_rndis.c
index 6614490..d2f1b6f 100644
--- a/drivers/usb/gadget/f_rndis.c
+++ b/drivers/usb/gadget/f_rndis.c
@@ -767,11 +767,11 @@ rndis_bind(struct usb_configuration *c, struct usb_function *f)
 
 	rndis_set_param_medium(rndis->config, NDIS_MEDIUM_802_3, 0);
 	rndis_set_host_mac(rndis->config, rndis->ethaddr);
-
+/*
 	if (rndis_set_param_vendor(rndis->config, vendorID,
 				manufacturer))
 		goto fail;
-
+*/
 	/* NOTE:  all that is done without knowing or caring about
 	 * the network link ... which is unavailable to this code
 	 * until we're activated via set_alt().
-- 
1.7.9.5

