From 92c8dd2fe6e62db7bcb07a9010c5eb9b757f2446 Mon Sep 17 00:00:00 2001
From: Joel A Fernandes <joelagnel@ti.com>
Date: Mon, 28 Nov 2011 20:56:48 -0600
Subject: [PATCH 11/26] meta-ti: Remove debug messages for meta-ti

Signed-off-by: Joel A Fernandes <joelagnel@ti.com>
---
 drivers/input/touchscreen/ti_tscadc.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/touchscreen/ti_tscadc.c b/drivers/input/touchscreen/ti_tscadc.c
index 8661b3a..c82f9bf 100644
--- a/drivers/input/touchscreen/ti_tscadc.c
+++ b/drivers/input/touchscreen/ti_tscadc.c
@@ -186,14 +186,14 @@ static irqreturn_t tsc_adc_interrupt(int irq, void *dev)
 
 	status = tscadc_readl(ts_dev, TSCADC_REG_IRQSTATUS);
 
-	printk("interrupt! status=%x\n", status);
+	// printk("interrupt! status=%x\n", status);
 	// if (status & TSCADC_IRQENB_EOS) {
 	// 	irqclr |= TSCADC_IRQENB_EOS;
 	// }
 
 	if (status & TSCADC_IRQENB_FIFO0THRES) {
 		fifo1count = tscadc_readl(ts_dev, TSCADC_REG_FIFO0CNT);
-		printk("fifo 0 count = %d\n", fifo1count);
+		// printk("fifo 0 count = %d\n", fifo1count);
 	
 		for (i = 0; i < fifo1count; i++) {
 			read_sample = tscadc_readl(ts_dev, TSCADC_REG_FIFO0);
-- 
1.7.9.5

