From 7c43a6da69624b297623916f780a1323f1af7c4e Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Sat, 19 May 2012 11:29:14 +0200
Subject: [PATCH 049/102] ts_tscadc: add defines for 4x and 16x oversampling

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 drivers/input/touchscreen/ti_tscadc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/input/touchscreen/ti_tscadc.c b/drivers/input/touchscreen/ti_tscadc.c
index 4fd2e18..acb62d1 100644
--- a/drivers/input/touchscreen/ti_tscadc.c
+++ b/drivers/input/touchscreen/ti_tscadc.c
@@ -83,6 +83,8 @@ static DEVICE_ATTR(ain8, S_IRUGO, do_adc_sample, NULL);
 #define TSCADC_STEPCONFIG_MODE_SWCONT		0x1
 #define TSCADC_STEPCONFIG_MODE_SWONESHOT	0x0
 #define TSCADC_STEPCONFIG_2SAMPLES_AVG	(1 << 4)
+#define TSCADC_STEPCONFIG_4SAMPLES_AVG	(1 << 3)
+#define TSCADC_STEPCONFIG_16SAMPLES_AVG	(1 << 2)
 #define TSCADC_STEPCONFIG_NO_AVG	0
 #define TSCADC_STEPCONFIG_XPP		BIT(5)
 #define TSCADC_STEPCONFIG_XNN		BIT(6)
-- 
1.8.1.4

