From f0917a423d44194557926634e6dfeaa969f8b758 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Sat, 19 May 2012 11:37:21 +0200
Subject: [PATCH 050/102] ts_tscadc: switch to 4x oversampling

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 drivers/input/touchscreen/ti_tscadc.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/input/touchscreen/ti_tscadc.c b/drivers/input/touchscreen/ti_tscadc.c
index acb62d1..7df7bb2 100644
--- a/drivers/input/touchscreen/ti_tscadc.c
+++ b/drivers/input/touchscreen/ti_tscadc.c
@@ -157,11 +157,11 @@ static void tsc_adc_step_config(struct tscadc *ts_dev, int channel)
 	/*
  	 * Step Configuration
  	 * software-enabled continous mode
- 	 * 2 sample averaging
+ 	 * 4 sample averaging
  	 * sample channel 1 (SEL_INP mux bits = 0)
  	 */
 	stepconfig = TSCADC_STEPCONFIG_MODE_SWONESHOT |
-		TSCADC_STEPCONFIG_2SAMPLES_AVG |
+		TSCADC_STEPCONFIG_4SAMPLES_AVG |
 		((channel-1) << 19);
 	
 	delay = TSCADC_STEPCONFIG_SAMPLEDLY | TSCADC_STEPCONFIG_OPENDLY;
@@ -242,7 +242,7 @@ static void tsc_step_config(struct tscadc *ts_dev)
 	delay = TSCADC_STEPCONFIG_SAMPLEDLY | TSCADC_STEPCONFIG_OPENDLY;
 
 	stepconfigx = TSCADC_STEPCONFIG_MODE_HWSYNC |
-			TSCADC_STEPCONFIG_2SAMPLES_AVG | TSCADC_STEPCONFIG_XPP;
+			TSCADC_STEPCONFIG_4SAMPLES_AVG | TSCADC_STEPCONFIG_XPP;
 
 	switch (ts_dev->wires) {
 	case 4:
@@ -279,7 +279,7 @@ static void tsc_step_config(struct tscadc *ts_dev)
 	}
 
 	stepconfigy = TSCADC_STEPCONFIG_MODE_HWSYNC |
-			TSCADC_STEPCONFIG_2SAMPLES_AVG | TSCADC_STEPCONFIG_YNN |
+			TSCADC_STEPCONFIG_4SAMPLES_AVG | TSCADC_STEPCONFIG_YNN |
 			TSCADC_STEPCONFIG_INM | TSCADC_STEPCONFIG_FIFO1;
 	switch (ts_dev->wires) {
 	case 4:
@@ -324,7 +324,7 @@ static void tsc_step_config(struct tscadc *ts_dev)
 
 	 /* Configure to calculate pressure */
 	stepconfigz1 = TSCADC_STEPCONFIG_MODE_HWSYNC |
-				TSCADC_STEPCONFIG_2SAMPLES_AVG |
+				TSCADC_STEPCONFIG_4SAMPLES_AVG |
 				TSCADC_STEPCONFIG_XNP |
 				TSCADC_STEPCONFIG_YPN | TSCADC_STEPCONFIG_INM;
 	stepconfigz2 = stepconfigz1 | TSCADC_STEPCONFIG_Z1 |
-- 
1.8.1.4

