From e8110997878f33f93c980c0e2fd54e55450b1d68 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Mon, 28 May 2012 09:56:45 +0200
Subject: [PATCH 65/92] ti_tscadc: switch to 16x averaging

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 drivers/input/touchscreen/ti_tscadc.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/input/touchscreen/ti_tscadc.c b/drivers/input/touchscreen/ti_tscadc.c
index 7df7bb2..ae6c4c4 100644
--- a/drivers/input/touchscreen/ti_tscadc.c
+++ b/drivers/input/touchscreen/ti_tscadc.c
@@ -161,7 +161,7 @@ static void tsc_adc_step_config(struct tscadc *ts_dev, int channel)
  	 * sample channel 1 (SEL_INP mux bits = 0)
  	 */
 	stepconfig = TSCADC_STEPCONFIG_MODE_SWONESHOT |
-		TSCADC_STEPCONFIG_4SAMPLES_AVG |
+		TSCADC_STEPCONFIG_16SAMPLES_AVG |
 		((channel-1) << 19);
 	
 	delay = TSCADC_STEPCONFIG_SAMPLEDLY | TSCADC_STEPCONFIG_OPENDLY;
@@ -242,7 +242,7 @@ static void tsc_step_config(struct tscadc *ts_dev)
 	delay = TSCADC_STEPCONFIG_SAMPLEDLY | TSCADC_STEPCONFIG_OPENDLY;
 
 	stepconfigx = TSCADC_STEPCONFIG_MODE_HWSYNC |
-			TSCADC_STEPCONFIG_4SAMPLES_AVG | TSCADC_STEPCONFIG_XPP;
+			TSCADC_STEPCONFIG_16SAMPLES_AVG | TSCADC_STEPCONFIG_XPP;
 
 	switch (ts_dev->wires) {
 	case 4:
@@ -279,7 +279,7 @@ static void tsc_step_config(struct tscadc *ts_dev)
 	}
 
 	stepconfigy = TSCADC_STEPCONFIG_MODE_HWSYNC |
-			TSCADC_STEPCONFIG_4SAMPLES_AVG | TSCADC_STEPCONFIG_YNN |
+			TSCADC_STEPCONFIG_16SAMPLES_AVG | TSCADC_STEPCONFIG_YNN |
 			TSCADC_STEPCONFIG_INM | TSCADC_STEPCONFIG_FIFO1;
 	switch (ts_dev->wires) {
 	case 4:
@@ -324,7 +324,7 @@ static void tsc_step_config(struct tscadc *ts_dev)
 
 	 /* Configure to calculate pressure */
 	stepconfigz1 = TSCADC_STEPCONFIG_MODE_HWSYNC |
-				TSCADC_STEPCONFIG_4SAMPLES_AVG |
+				TSCADC_STEPCONFIG_16SAMPLES_AVG |
 				TSCADC_STEPCONFIG_XNP |
 				TSCADC_STEPCONFIG_YPN | TSCADC_STEPCONFIG_INM;
 	stepconfigz2 = stepconfigz1 | TSCADC_STEPCONFIG_Z1 |
-- 
1.7.7.6

